torch_spyre/_inductor/stickify.py

```diff
@@ -93,124 +106,208 @@ def get_allocation_size(self) -> list[Expr]:
+        return FixedTiledLayout(
+            output.device, output.dtype, output.size, output.stride, stl
+        )
     else:
-        raise Unsupported(
-            f"binop with incompatible DCIs: {x_layout} {y_layout} {x_broadcasted} {y_broadcasted}"
+        input = args[0]
+        input_dims = stride_order_vars(input.dep.index)
+        stick_var = input_dims[-1]
+        is_stick_reduction = stick_var not in output_dims
+        stl = SpyreTensorLayout(output.size, output.dtype)
+        stl.format = StickFormat.Sparse if is_stick_reduction else StickFormat.Dense
+        return FixedTiledLayout(
+            output.device, output.dtype, output.size, output.stride, stl
         )
 
-    # TODO: Forcing generic stick dimension order
-    dim_order = list(range(len(res_size)))
-    return res_size, SpyreTensorLayout(res_size, x.dtype, dim_order, format=res_format)
+
+def propagate_spyre_tensor_layouts(
+    nodes: list[BaseSchedulerNode],
+) -> list[BaseSchedulerNode]:
+    def mem_deps(n: SchedulerNode) -> list[Arg]:
+        res: list[Arg] = []
+        for arg in n.read_writes.reads:
+            if isinstance(arg, MemoryDep):
+                buf = V.graph.get_buffer(arg.name)
+                layout = buf.get_layout()
+                if not isinstance(layout, FixedTiledLayout):
+                    raise RuntimeError(f"{buf} does not have FixedTiledLayout")
+                res.append(Arg(arg, layout))
+        return res
+
+    # Convert InputBuffers from FixedLayout to FixedTiledLayouts
+    for name, real_input in zip(V.graph.graph_input_names, V.get_real_inputs()):
+        if isinstance(real_input, torch.Tensor):
+            stl = real_input.device_tensor_layout()
+            if stl is None:
+                # TODO: This should become a hard-error
+                print(
+                    f"Warning: missing device_tensor_layout on graph input {name}; assuming generic stick layout"
+                )
+                stl = SpyreTensorLayout(real_input.size(), real_input.dtype)
+            tb = V.graph.graph_inputs[name]
+            if (
+                not isinstance(tb, TensorBox)
+                or not isinstance(tb.data, StorageBox)
+                or not isinstance(tb.data.data, InputBuffer)
+            ):
+                raise Unsupported(
+                    "graph input {name} is not a TensorBox(StorageBox(InputBuffer))"
+                )
+            ptl = tb.data.data.layout
+            if not isinstance(ptl, FixedLayout):
+                raise Unsupported("graph input {name} does not have a FixedLayout")
+            tb.data.data.layout = FixedTiledLayout(
+                ptl.device, ptl.dtype, ptl.size, ptl.stride, stl
+            )
+
+    # Nodes are in topological order (guarenteed by caller).
+    # Visit them and use the inputs' FixedTiledLayouts and the operation being
+    # performed by the node to convert its output FixedLayouts to FixedTiledLayouts.
```

Comment:
Do we realize every single tensor still with this refactor? Or are we allowing FlexibleLayouts at some point until we have to realize everything?